p11-kit/rpc-server.c

```diff
@@ -1175,7 +1086,7 @@ rpc_C_GetAttributeValue (CK_X_FUNCTION_LIST *self,
 	BEGIN_CALL (GetAttributeValue);
 		IN_ULONG (session);
 		IN_ULONG (object);
-		IN_ATTRIBUTE_BUFFER (template, count);
```

Comment:
I'm not sure if it is a good idea to replace "buffer" with "array". The difference between those is that "array" needs all elements to be known and initialized prior to the call, while "buffer" can contain uninitialized element, e.g., `attr->pValue` is set to NULL to retrieve `attr->ulValueLen`.

In any case, their RPC signatures are also different, so if we change "buffer" to "array", we might need a version bump of the RPC protocol and a proper fallback to the previous behavior.

Reply:
I have kept the IN_ATTRIBUTE_BUFFER and made it to work with recursive attributes.